packages/block-library/src/columns/index.js

```diff
@@ -155,7 +164,7 @@ export const settings = {
 	],
 
 	edit( { attributes, setAttributes, className } ) {
-		const { columns } = attributes;
+		const columns = attributes.columns || DEFAULT_NUMBER_OF_COLUMNS;
```

Comment:
Begs the question: Should a block developer ever need to anticipate an attribute value being `undefined` when its definition includes a `default` ? Is this something which ought to be handled when setting attributes? Probably a decent amount of overhead included in doing so. Alternatively it could be block specific in only letting the `RangeControl`'s `onChange` trigger when a valid `nextColumns`, or provide the defaulting there, or `RangeControl` provides a fallback value when its input is invalid.

Thoughts?

Reply:
Hi @aduth, you are comments are right, given that a block has a default we should not need to anticipate an undefined value.
But setting changing the attribute to the default when an undefined value exists causes some problems. E.g: a user has 3 columns wants to set to two 4 columns, deletes the 3 and in the UI automatically sees a 2 instead of an empty field to write the 3.

This problem is resolved with a PR I created that fixes some behaviors in the RangeControl https://github.com/WordPress/gutenberg/pull/12952.
If these changes are merged after the other PR everything will work smoothly.